tests/test_buffers.py

```diff
@@ -239,3 +239,117 @@ def test_buffer_exception():
         memoryview(m.BrokenMatrix(1, 1))
     assert isinstance(excinfo.value.__cause__, RuntimeError)
     assert "for context" in str(excinfo.value.__cause__)
+
+
+def test_to_pybuffer():
+    mat = m.Matrix(5, 4)
```

Comment:
> To compare with NumPy, you can swap:
> - `Matrix(5, 4)` with `np.empty((5, 4), dtype=np.float32)`
> - `FortranMatrix(5, 4)` with `np.empty((5, 4), dtype=np.float32, order='F')`
> - `DiscontiguousMatrix(5, 4, 2, 3)` with `np.empty((5*2, 4*3), dtype=np.float32)[::2, ::3]`
> Also replace `mat.rows()` with 5 and `mat.cols()` with 4 (maybe I should do that so we're always comparing with 'known' values instead of internals?)

> Then running the tests show two differences:
> - for `PyBUF_SIMPLE`, NumPy returns `ndim=0`, which I guess is truer in spirit, and we could do here too.
> - for contiguity mismatches, it raises `ValueError` instead of `BufferError`, which is maybe a bug on their part, but close enough to the same.

Could you please try that? Roughly:

- Split the added tests into multiple smaller ones.
- Use `@pytest.mark.parametrize` to loop over the `np.empty()` equivalents above.

Motivation behind my suggestion:
- We want to ensure that we're compatible with numpy, and that it stays that way in the future.

(I think you can use `pytest.importorskip("numpy")`, although we only want to skip the tests that depend on numpy (not all tests in this file). Not sure exactly how to best make that work.)

Reply:
> (I think you can use `pytest.importorskip("numpy")`, although we only want to skip the tests that depend on numpy (not all tests in this file). Not sure exactly how to best make that work.)

No need; it's already at the top of the file.